akka-stream/src/main/scala/akka/stream/BufferStatus.scala

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream
+
+class BufferStatus(val used: Int, val capacity: Int) {
+  def getUsed() = used
+  def getCapacity() = capacity
+}
+
+sealed abstract class GetStatusMessage
+
+object GetStatusMessage {
+  val instance = new GetStatusMessage {}
+
+  def apply(): GetStatusMessage = instance
+
+  def getStatusMessage(): GetStatusMessage = instance
+}
```

Comment:
we typically define such singleton objects as:

```scala
@DoNotInherit sealed abstract class GetStatusMessage

case object GetStatusMessage extends GetStatusMessage {
  /**
   * Java API: the singleton instance
   */
  def getInstance(): GetStatusMessage = this
}
```

and when matching `case GetStatusMessage =>`